include/nil/crypto3/zk/snark/arithmetization/plonk/padding.hpp

```diff
@@ -76,6 +79,68 @@ namespace nil {
                     return padded_rows_amount;
                 }
 
+
+                template<typename FieldType, typename ArithmetizationParams, typename ColumnType>
+                std::uint32_t zk_padding(
+                    plonk_table<FieldType, ArithmetizationParams, ColumnType> &table,
+                    typename nil::crypto3::random::algebraic_engine<FieldType> alg_rnd = nil::crypto3::random::algebraic_engine<FieldType>()
+                ) {
+                    std::uint32_t usable_rows_amount = table.rows_amount();
+
+                    std::uint32_t padded_rows_amount = std::pow(2, std::ceil(std::log2(usable_rows_amount)));
```

Comment:
If I understand correctly, we are pre-filling the whole table with random values, such that whatever is unused will have random values. If we could do the opposite, fill whatever is going to be used, and then fill the rest with random values, maybe that would be better. But this is also ok...